### PR TITLE
Upgrade to Firebase 3

### PR DESCRIPTION
**Pull request by simenbrekken**
While this is functionally equivalent to the 2.x version there's a few outstanding issues that needs attention:

- [x] Examples
- [x] Docs

### CODE REVIEW

**Comment by simenbrekken** [comment]
@einarlove When you've got some spare time, could you look over this and see if it checks out?

File: README.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 React Firebase
 ==============
 
-React bindings for [Firebase](https://www.firebase.com).
+React bindings for [Firebase](https://firebase.google.com).
 
 ## Installation
 
-React Firebase requires **React 0.14 or later.**
+React Firebase requires **React 15 and Firebase 3 or later.**
```
Comment (einarlove) [inline README.md]: Why doesn't it support react 0.14?

**Comment by simenbrekken** [comment]
Still need to fine-tune the API to make it as easy as the first version. 

A great suggestion from @einarlove is to change the signature of `mapFirebaseToProps` to `mapFirebaseToProps(ref, app)` where `ref = firebase.database().ref()` and `app = firebase.initializeApp()` so that we have easy access to the root ref while still being able to access other parts of the namespace such as `auth` and `storage`

**Comment by einarlove** [comment]
Suggestion for another API design for `connect`
```js
export default connect((props, ref, { auth }) => ({
    greetings: 'greetings',
    people: ref('people').limitToFirst(props.limit),
    profile: `profiles/${firebase.auth().id}`,
  }

}, (props, ref, { auth }) => {
  addGreeting: greeting => ref('greeting').push(greeting)
})(GreetingsList)
```

**Comment by simenbrekken** [comment]
@einarlove I switched the arguments order to make sure `ref` always comes first as many queries (e.g. `todos`) might not require props.

File: package.json [hunk 2]
```diff
@@ -1,20 +1,33 @@
 {
   "name": "react-firebase",
-  "version": "0.1.0",
+  "version": "2.0.0",
   "description": "React bindings for Firebase",
-  "main": "./lib/index.js",
+  "author": "Simen Brekken <simen@unfold.no> (http://github.com/unfold)",
```
Comment (einarlove) [inline package.json]: Our current convention is to use company as author
- [webpack-serve/package.json#L27](https://github.com/unfold/webpack-serve/blob/master/package.json#L27)
- [fetch-data/package.json#L27](https://github.com/unfold/fetch-data/blob/master/package.json#L27)
- [config-variables/package.json#L21](https://github.com/unfold/config-variables/blob/master/package.json#L21)

**Comment by einarlove** [comment]
I speculate that props will be more used than ref and thus should be the first argument. This could be resolved if we stopped passing `ref` in connect and gave the developer option to generate query with an object instead
```js
connect(props => ({
  posts: `companies/${props.company}/posts`,
  users: {
    path: `companies/${props.company}/users`,
    orderByChild: 'age',
    limitToLast: props.showAll ? null : 30,
  },
}))(Component)
```

**Comment by einarlove** [comment]
Example of combined `mapPropsToSubscriptions` `mapFirebaseToProps`
```js
connect((props, ref) => ({
  users: `companies/${props.company}/users`,
  posts: {
    path: `companies/${props.company}/posts`,
    orderByChild: 'createdAt',
    limitToFirst: props.showAll ? null : props.pageSize,
  },
  createPost: post => ref(`companies/${props.company}/posts`).push(post),
}))(Component)
```

File: src/components/connect.js [hunk 3]
```diff
@@ -1,117 +1,117 @@
...
+  const finalMergeProps = mergeProps || defaultMergeProps
+  const { pure = true, keepalive = 0 } = options
 
-  const finalMergeProps = mergeProps || defaultMergeProps;
-  const { pure = true, keepalive = 0 } = options;
-
-  const computeSubscriptions = props => {
-    const subscriptions = mapSubscriptions(props);
+  const computeSubscriptions = (props, firebase) => {
+    const ref = path => firebase.database().ref(path)
```
Comment (einarlove) [inline src/components/connect.js]: Why not
```const subscriptions = mapSubscriptions(firebase.database().ref, props, firebase)```

File: package.json (hunk 2 above)
Comment (einarlove) [inline package.json]: Still sticking with your name?

Comment (simenbrekken) [inline package.json]: @einarlove Fixed in https://github.com/unfold/react-firebase/pull/6/commits/ffe5c69b066e419c7e476de8b22f8385c3fcf130

I think we should expand the author to include email and github page like in that commit

File: README.md [hunk 4]
```diff
@@ -1,61 +1,27 @@
 React Firebase
 ==============
 
-React bindings for [Firebase](https://www.firebase.com).
+React bindings for [Firebase](https://firebase.google.com).
 
 ## Installation
 
-React Firebase requires **React 0.14 or later.**
+React Firebase requires **React 0.14 and Firebase 3 or later.**
```
Comment (einarlove) [inline README.md]: Maybe link to these packages?

File: README.md [hunk 5]
```diff
@@ -1,61 +1,27 @@
...
 
 ```
 npm install --save react-firebase
 ```
 
 ## Usage
```
Comment (einarlove) [inline README.md]: The documentation is quite heavy and detailed, so I would suggest having a simple table of contents at the top, or/and give a simple example before diving in to Static Properties, arguments etc.

File: README.md [hunk 6]
```diff
@@ -67,65 +33,80 @@ A React component class that injects subscriptions and actions into your compone
 
 #### Remarks
 
-* It needs to be invoked two times. The first time with its arguments described above, and a second time, with the component: `connect(mapPropsToSubscriptions, mapFirebaseToProps, mergeProps)(MyComponent)`.
+* It needs to be invoked two times. The first time with its arguments described above, and a second time, with the component: `connect(mapFirebaseToProps)(MyComponent)`.
```
Comment (einarlove) [inline README.md]: What? Does this effect the developer or component in a way we should be vary about?

File: README.md [hunk 7]
```diff
@@ -67,65 +33,80 @@ A React component class that injects subscriptions and actions into your compone
...
 
 * It does not modify the passed React component. It returns a new, connected component, that you should use instead.
 
 #### Examples
 
 > Runnable examples can be found in the [examples folder](examples/).
 
-##### Inject `firebase` and `todos`
+##### Inject `todos`
```
Comment (einarlove) [inline README.md]: Injects feels kinda foreign to me.
Maybe **# Pass todos as prop**

File: README.md [hunk 8]
```diff
@@ -67,65 +33,80 @@ A React component class that injects subscriptions and actions into your compone
...
 
   > Note: The value of `todos` is analogous to https://my-firebase.firebaseio.com/todos.
```
Comment (einarlove) [inline README.md]: analogous is an uncommon word.

Perhaps:
> Note: `'todos'` is the path to the resource. Same as using `firebase.database().ref('todos')` or https://my-firebase.firebaseio.com/todos

File: README.md (hunk 5 above)
Comment (einarlove) [inline README.md]: Would also suggest adding a section for `# Examples` which describes the examples and how to run them.

Comment (einarlove) [inline README.md]: A simple example should give you a glance of what you could do with it. I Often start using a lib before reading a docs if the example is short and good enough.

```js
import firebase from 'firebase'
import { connect } from 'react-firebase'

firebase.initializeApp({
  databaseURL: 'https://react-firebase-sandbox.firebaseio.com'
})

const Counter = ({ value, setValue }) => (
  <div>
    <button onClick={() => setCount(value - 1)}>-</button>
    <span>{value}</span>
    <button onClick={() => setCount(value + 1)}>+</button>
  </div>
)

export default connect((props, ref) => ({
  value: 'counterValue',
  setValue: value => ref('counterValue').set(value)
}))(TodosApp)
```

File: README.md (hunk 4 above)
Comment (einarlove) [inline README.md]: Think the requirements should come after the installation command.

File: README.md [hunk 9]
```diff
@@ -67,65 +33,80 @@ A React component class that injects subscriptions and actions into your compone
...
+### `<Provider firebaseApp>`
+
+By default `connect()` will use the [default Firebase App](https://firebase.google.com/docs/reference/js/firebase.app). If you have multiple Firebase App references in your application you may use this to specify the Firebase App reference available to `connect()` calls in the component hierarchy below.
+
+If you *really* need to, you can manually pass `firebaseApp` as a prop to every `connect()`ed component, but we only recommend to do this for stubbing `firebaseApp` in unit tests, or in non-fully-React codebases. Normally, you should just use `<Provider>`.
+
+#### Props
+
+* `firebaseApp` (*[App](https://firebase.google.com/docs/reference/js/firebase.app.App)*): A Firebase App reference.
+* `children` (*ReactElement*) The root of your component hierarchy.
```
Comment (einarlove) [inline README.md]: Missing `:` after `(*ReactElement*)`

Comment (simenbrekken) [inline README.md]: Fixed in https://github.com/unfold/react-firebase/commit/4abe20198448fd1c27525ac0c24d0d9fdd97ebf7

File: README.md (hunk 6 above)
Comment (simenbrekken) [inline README.md]: Fixed inhttps://github.com/unfold/react-firebase/commit/352837d45b440e61d9f9063579bb32eb88698a96

File: README.md (hunk 7 above)
Comment (simenbrekken) [inline README.md]: Fixed in 82cec8b19a4db9d7c03cae8910f7e3ae82738c04

File: README.md [hunk 10]
```diff
@@ -3,6 +3,30 @@ React Firebase
 
 React bindings for [Firebase](https://firebase.google.com).
 
+## Quick Start
+
+```
```
Comment (einarlove) [inline README.md]: add js syntax

File: README.md (hunk 4 above)
Comment (simenbrekken) [inline README.md]: Fixed in fd245f9f225684b376400dae048b1acceb1a6398

File: README.md (hunk 8 above)
Comment (simenbrekken) [inline README.md]: Fixed in 70c0582c2cf2a226dccb5277ffead42a432110be

File: README.md (hunk 10 above)
Comment (simenbrekken) [inline README.md]: Fixed in 4c559d37d8a930dc0a55feded3fdac78c98293aa

**Pull request merged**